### PR TITLE
Save the current viewMode in localStorage

### PR DESCRIPTION
**Pull request by hkhanouchi**
#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

### CODE REVIEW

File: src/zns-route-connect.test.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Container } from './zns-route-connect';
 import { Main } from './Main';
 import { Web3Connect } from './components/web3-connect';
 import { Authentication } from './components/authentication';
+import { ViewModes } from './shared-components/theme-engine';
```
Comment (mrval5) [inline src/zns-route-connect.test.tsx:7]: can you remove this line

File: src/store/theme/saga.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { takeLatest, put } from 'redux-saga/effects';
+import { SagaActionTypes, receive } from '.';
+import { ViewModes } from '../../shared-components/theme-engine';
+
+export const savedViewMode = 'viewMode:isLight';
+
+export function* setViewMode(action) {
+  const viewMode = action.payload;
+
+  localStorage.setItem(savedViewMode, viewMode === ViewModes.Dark ? 'false' : 'true');
```
Comment (mrval5) [inline src/store/theme/saga.ts:10]:
```suggestion
  localStorage.setItem(savedViewMode, viewMode === ViewModes.Dark);
```

File: src/store/theme/saga.test.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      },
+    };
+  });
+
+  it('sets viewMode theme', async () => {
+    const {
+      storeState: { value },
+    } = await expectSaga(setViewMode, { payload: ViewModes.Light }).withReducer(reducer).run();
+
+    expect(value.viewMode).toEqual(ViewModes.Light);
```
Comment (mrval5) [inline src/store/theme/saga.test.ts:28]: can we test on localStorage if it's called with the right params

File: src/components/theme-engine/index.tsx [hunk 4]
```diff
@@ -22,9 +25,18 @@ export class Container extends React.Component<Properties> {
...
-    return {};
+    return { setViewMode };
   }
 
+  componentDidMount() {
+    const viewMode = this.getSavedViewMode(savedViewMode) === 'true' ? ViewModes.Light : ViewModes.Dark;
+    this.props.setViewMode(viewMode);
+  }
+
+  getSavedViewMode = (savedViewMode: string): string => {
```
Comment (mrval5) [inline src/components/theme-engine/index.tsx:36]: this method returns either true or false as string, which is not what the name of the function describe.

can wee change it to `isLightModeIsSaved` (this is just an example). also to get rid of the logic of `'true'` in `componentDidMount`

File: src/store/theme/saga.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { takeLatest, put } from 'redux-saga/effects';
+import { SagaActionTypes, receive } from '.';
+import { ViewModes } from '../../shared-components/theme-engine';
+
+export const savedViewMode = 'viewMode:isLight';
```
Comment (mrval5) [inline src/store/theme/saga.ts:5]:
```suggestion
export const keyStorageLightMode = 'viewMode:isLight';
```

**Review by mrval5 — CHANGES_REQUESTED** [review]

File: src/store/theme/saga.ts (hunk 2 above)
Comment (hkhanouchi) [reply src/store/theme/saga.ts:10]: Argument of type 'boolean' is not assignable to parameter of type 'string'

**Pull request merged**